addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,420 @@
+from odoo.exceptions import RedirectWarning, UserError
+import logging
+import requests
+
+_logger = logging.getLogger(__name__)
+DEFAULT_ENDPOINT = "https://gw.fbr.gov.pk/DigitalInvoicing/v1/PostInvoiceData_v1"
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = "account.edi.format"
+
+    def _is_enabled_by_default_on_journal(self, journal):
+        self.ensure_one()
+        if self.code == "pk_einvoice_1_3":
+            return False
+        return super()._is_enabled_by_default_on_journal(journal)
+
+    def _get_l10n_pk_sales_group_ids(self, company_id):
+        return (
+            self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_0')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_1')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_2')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_3')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_5')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_8')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_10')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_13')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_15')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_16')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_17')
+            + self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_195')
+        ).ids
+
+    def _get_l10n_pk_withhold_group_ids(self, company_id):
+        return (
+            self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_wt')
+        ).ids
+
+    def _get_move_applicability(self, move):
+        # EXTENDS account_edi
+        self.ensure_one()
+        if self.code != 'pk_einvoice_1_3':
+            return super()._get_move_applicability(move)
+        is_taxeble = any(line.tax_group_id.id in self._get_l10n_pk_sales_group_ids(move.company_id) for line in move.line_ids.tax_ids)
+        if move.is_sale_document(include_receipts=True) and move.country_code == 'PK' and is_taxeble:
+            return {
+                'post': self._l10n_pk_edi_post_invoice,
+                'cancel': self._l10n_pk_edi_cancel,
+                'edi_content': self._l10n_pk_edi_xml_invoice_content,
+            }
+
+    def _check_move_configuration(self, move):
+        if self.code != "pk_einvoice_1_3":
+            return super()._check_move_configuration(move)
+        error_message = []
+        error_message += self._l10n_pk_validate_partner(move.partner_id)
+        error_message += self._l10n_pk_validate_partner(move.company_id.partner_id, is_company=True)
```

Comment:
```suggestion
```
is_company seems to be unused param. Can you explain it and why calling the method twice?

Reply:
I used the method twice to validate the company and the customer. Are we not validating a customer (`partner_id`)?